### PR TITLE
build (upgrade) edc 0.8.2-SNAPSHOT version

### PR DESCRIPTION
**Pull request by ahmetuyanikdev**
## What this PR changes/adds
Modified some OBS classes to adapt the project based on  edc version 0.8.2-snapshot 
Modified some test classes to adapt and build the project based on edc version 0.8.2-snapshot
Modified tests under e2e folder to run smooth unit and integration tests
Modified .toml gradle dependeny file for new version of edc
## Why it does that

EDC version upgrade for Huawei project

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

**Comment by ahmetuyanikdev** [comment]
@mspiekermann @janpmeyer this is draft pr for edc version upgrade

File: build.gradle.kts [hunk 1]
```diff
@@ -34,14 +34,11 @@ allprojects {
...
     configure<org.eclipse.edc.plugins.autodoc.AutodocExtension> {
         processorVersion.set(annotationProcessorVersion)
-        outputDirectory.set(project.buildDir)
+        outputDirectory.set(project.layout.buildDirectory.asFile.get())
     }
 
     configure<org.eclipse.edc.plugins.edcbuild.extensions.BuildExtension> {
-        versions {
-            // override default dependency versions here
-            metaModel.set(metaModelVersion)
```
Comment (ndr-brt) [inline build.gradle.kts:43]: the `metaModelVersion` can be removed from this file and from `gradle.properties` as well

File: build.gradle.kts [hunk 2]
```diff
@@ -34,14 +34,11 @@ allprojects {
     // configure which version of the annotation processor to use. defaults to the same version as the plugin
     configure<org.eclipse.edc.plugins.autodoc.AutodocExtension> {
         processorVersion.set(annotationProcessorVersion)
-        outputDirectory.set(project.buildDir)
+        outputDirectory.set(project.layout.buildDirectory.asFile.get())
     }
```
Comment (ndr-brt) [inline build.gradle.kts:38]: this autodoc block is not needed as by default it is configured this way.
the `annotationProcessorVersion` could then be removed by both this file and `gradle.properties`

another thing to do is to use project `version` as `edc-build` plugin version, then `edcGradlePluginsVersion` can be removed both from this file and from `gradle.properties`

you can get some inspiration from another technology repo: https://github.com/eclipse-edc/Technology-Aws/blob/main/build.gradle.kts

File: gradle.properties [hunk 3]
```diff
@@ -13,11 +13,11 @@
...
 group=com.huawei.cloud
-version=0.0.1-SNAPSHOT
+version=0.8.2-SNAPSHOT
 # configure the build:
-annotationProcessorVersion=0.5.0
-edcGradlePluginsVersion=0.5.0
-metaModelVersion=0.5.0
+annotationProcessorVersion=0.8.2-SNAPSHOT
+edcGradlePluginsVersion=0.8.2-SNAPSHOT
+metaModelVersion=0.8.2-SNAPSHOT
```
Comment (ndr-brt) [inline gradle.properties:20]: as reported in other comments, these variables can be removed

**Pull request closed without merging**